bin/validate.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+import jsonschema
+from jsonschema import validate
+import yaml
+from yaml import load, dump, Loader, Dumper
+import click
+import os
+import sys
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+
+def load_doc(filename):
+    with open(filename, "r") as stream:
+    	try:
+        		return yaml.safe_load(stream)
+    	except yaml.YAMLError as exc:
+        		print(exc)
+
+def validate_config(config):
+    try:
+        validate(instance=config, schema=schema)
+    except jsonschema.exceptions.ValidationError as err:
+        print(err, file=sys.stderr)        
+        return False
+    return True
+
+def check_slot(adata, slot_type, slot_name):
+
+    print(f"Checking for {slot_type} {slot_name}")
+
+    if slot_type == 'matrices':
+        if slot_name == 'X':
+            return hasattr(adata, 'X')
+        elif slot_name == 'raw.X':
+            return hasattr(adata.raw, 'X')
+        else:
+            return slot_name in adata.layers
+    
+    elif slot_type == 'dimension_reductions':
+        return slot_name in adata.obsm
+
+    elif slot_type == 'cell_groups':
+        return slot_name in adata.obs.columns
+
+    else:
+        print(f"{slot_type} slot type not recognised", file=sys.stderr)
+        return False 
+
+schema_file = os.path.join(script_dir, '..', 'config_schema.yaml')
+schema = load_doc(schema_file) 
```

Comment:
I would suggest maybe packing the schema with the pip package (and reading the file from the package).

Reply:
There's no package here yet, but will do!